src/utils/diagnostic.ts

```diff
@@ -0,0 +1,70 @@
+import { Diagnostic, DiagnosticSeverity, Range } from 'vscode-languageserver';
+
+const ParseErrorExp = /^Parse error on line (\d+)/;
+const OnLineErrorExp = / \(on line (\d+)\)\.$/;
+
+export interface TemplateLinterError {
+  moduleId: string;
+  message: string;
+  source: string;
+  line?: number;
+  column?: number;
+}
+
+export function toDiagnostic(source: string, error: TemplateLinterError): Diagnostic {
+  return {
+    severity: DiagnosticSeverity.Error,
+    range: toRange(source, error),
+    message: toMessage(error),
+    source: 'ember-template-lint'
```

Comment:
So you want you want a different `source` depending on the actual source of the error? I would argue, that thats rather confusing.  `ember-template-lint` is throwing the error. If it is because of an syntax error or caused by a linter rule should not matter to the user. But if you prefer the source to reflect that I can change it.